x-pack/plugins/maps/public/embeddable/map_embeddable.tsx

```diff
@@ -137,6 +137,36 @@ export class MapEmbeddable
     this._savedMap = new SavedMap({ mapEmbeddableInput: initialInput });
     this._initializeSaveMap();
     this._subscription = this.getUpdated$().subscribe(() => this.onUpdate());
+    this.setEventHandlers({
```

Comment:
Instead of updating `loading` and `error` via event handlers, which are low level and not implemented for all layer types (like vector tiles), how about updating `_handleStoreChanges` to update `loading` and `error` based on if any layers return true for `layer.isLayerLoading` or `layer.hasErrors`. You can get the layers from the store with the redux selector `getLayerList`

Reply:
Would you mind making a `patch` for this? 
I'm not super familiar with the maps code.

Reply:
Created https://github.com/lizozom/kibana/pull/9